### PR TITLE
Searchv3: Profile view on mobile

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

Here's the searchv3 profile implementation on mobile.  I haven't tested it fully because ResultList doesn't exist for mobile yet.  And I'll be out next week, so likely someone else will want to take over this PR.

I think it would be safe to merge this now -- if you want to -- before ResultList lands, since this uses the searchv3 feature flag.

(This splits the container out into its own file, and creates separate desktop and native render components.  So the actual changes are very small, even though the diff ended up large.)

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits bac4e625029a...3a97ccbbc3a7 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497041184784-bac4e625029a-3a97ccbbc3a7-linux/Tracker-New_user_Scroll_1-bac4e625029a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497041184784-bac4e625029a-3a97ccbbc3a7-linux/Tracker-New_user_Scroll_1-3a97ccbbc3a7.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

File: shared/profile/search-container.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  _onChangeText: (nextText: string) => void,
+  _onSelectService: () => void,
+  _search: (term: string, service: SearchConstants.Service) => void,
+}
+
+const mapStateToProps = (state: TypedState) => ({
+  ids: profileSearchResultArray(state),
+})
+const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, onBack, onToggleSidePanel}: Props) => ({
+  _clearSearchResults: id => dispatch(Creators.clearSearchResults()),
```
Comment (chromakode) [inline shared/profile/search-container.js]: :sheep: Is `id` here really a parameter? I expected this to have no arguments according to the type.

File: shared/profile/search.desktop.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      <Box style={{...globalStyles.flexBoxRow, flexGrow: 1}}>
+        <Box style={{flexGrow: 1}}>
+          <UserInput
+            userItems={props.userItems}
+            showAddButton={props.showAddButton}
+            onRemoveUser={props.onRemoveUser}
+            onClickAddButton={props.onClickAddButton}
+            placeholder={props.placeholder}
+            usernameText={props.searchText}
+            onChangeText={props.onChangeText}
```
Comment (chromakode) [inline shared/profile/search.desktop.js]: Now that my user input arrow keys commit is in, this needs no-ops for the handlers to pass flow: https://github.com/keybase/client/commit/eedb9739243d9c087f5519dfe8d0ef1369d7c350

I made these handlers required props so we wouldn't forget them. For native they never get called, and can be null or no-ops.

File: shared/profile/search.js.flow [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// @flow
+import {Component} from 'react'
+
+import type {SearchResultId} from '../constants/searchv3'
+import type {UserDetails} from '../searchv3/user-input'
+
+export type Props = {
+  ids: Array<SearchResultId>,
+  navigateUp: () => void,
+  onClick: () => void,
```
Comment (chromakode) [inline shared/profile/search.js.flow]: Doesn't this receive an `id` argument?

File: shared/profile/search.desktop.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import type {Props} from './search'
+
+const Search = (props: Props) => (
+  <Box style={styleSearchContainer}>
+    <Box style={{...styleSearchRow}}>
+      <Box style={{...globalStyles.flexBoxRow, flexGrow: 1}}>
+        <Box style={{flexGrow: 1}}>
+          <UserInput
+            userItems={props.userItems}
+            showAddButton={props.showAddButton}
```
Comment (chromakode) [inline shared/profile/search.desktop.js]: This prop is no longer used by the input.

File: shared/profile/search.native.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        placeholder={props.placeholder}
+        usernameText={props.searchText}
+        onChangeText={props.onChangeText}
+      />
+    </Box>
+    <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', justifyContent: 'center'}}>
+      <Text style={{marginRight: globalMargins.tiny}} type="BodySmall">Filter:</Text>
+      <ServiceFilter selectedService={props.selectedService} onSelectService={props.onSelectService} />
+    </Box>
+    <Box style={{...globalStyles.scrollable}}>
```
Comment (chromakode) [inline shared/profile/search.native.js]: It looks like `globalStyles.scrollable` doesn't exist for native?

**Comment by chrisnojima** [comment]
@chromakode want to take this over since you did this review?

**Comment by chromakode** [comment]
Updated! Please review ea9078b...c259663.

File: shared/profile/search.desktop.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import ResultsList from '../searchv3/results-list'
+import UserInput from '../searchv3/user-input'
+import {Box, Icon, Text} from '../common-adapters'
+import {globalStyles, globalColors, globalMargins} from '../styles'
+
+import type {Props} from './search'
+
+const Search = (props: Props) => (
+  <Box style={styleSearchContainer}>
+    <Box style={{...styleSearchRow}}>
```
Comment (chrisnojima) [inline shared/profile/search.desktop.js]: 🐑  can be just styleSearchRow

File: shared/searchv3/results-list/index.desktop.js [hunk 7]
```diff
@@ -29,15 +23,8 @@ class SearchResultsList extends Component<void, Props, void> {
 
   render() {
     const {style, items} = this.props
-
-    // TODO maybe move to container so this is shared
```
Comment (chrisnojima) [inline shared/searchv3/results-list/index.desktop.js]: i was thinking you could actually do the switching of empty results in a recompse branch() call and not have these components do the checking at all

**Pull request merged**